pkg/kvstore/allocator/allocator.go

```diff
@@ -778,16 +802,27 @@ func (a *Allocator) startWatch() waitChan {
 					case kvstore.EventTypeCreate, kvstore.EventTypeModify:
 						kvstore.Trace("Adding id to cache", nil, logrus.Fields{fieldKey: key, fieldID: id})
 						a.nextCache[id] = key
+						a.nextKeyCache[key.GetKey()] = id
 					case kvstore.EventTypeDelete:
 						kvstore.Trace("Removing id from cache", nil, logrus.Fields{fieldID: id})
 						delete(a.nextCache, id)
+
+						// key value is not available. Need to scan the cache
```

Comment:
It's not available because you have delete it in `808`.

Can't you do:
```
key := a.nextCache[id]
delete(a.nextKeyCache, key)
delete(a.nextCache, id)
```

Reply:
Good point. Will do.